plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts

```diff
@@ -126,6 +126,9 @@ export class DefaultFactRetrieverEngine implements FactRetrieverEngine {
             frequency: { cron: cronExpression },
             fn: this.createFactRetrieverHandler(factRetriever, lifecycle),
             timeout: timeLimit,
+            // We add a delay in order to prevent errors due to the
+            // fact that the backend is not yet online in a cold-start scenario
+            initialDelay: Duration.fromObject({ seconds: 5 }),
```

Comment:
Yeah. That's what I first thought as well. Up for discussion. I figured that this would be simple. No added options, thus just as simple as before. For jobs that are run at most every minute, a 5 second delay did not seem such a downside for me.

But of course it would be no problem to have this as an option in some way.

Reply:
Let's add this as an optional config value but default to a sane value if it's missing.
Other than that, this looks good to me.

Reply:
Thanks for the feedback. I made a change today that adds `initialDelay` as an option for a factRetrieverRegistration. It now default to the 5 seconds I proposed earlier.